### PR TITLE
feat(init-templates): app template comes with hint comments for 'env'

### LINKED ISSUE #12321: (aws-ec2): Vpc.from_lookup does not work out of the box on init templates
This sample code that shows the env settings seem to be ignored when the VPCImport2 class is ran. If I comment out the import_vpc_stack line, it runs fine and creates the VPC. But when using the vpc.from_lookup function is always fails. This code is just a representation of trying to use vpc.from_import and showing that the env actually works.

### Reproduction Steps
```
from aws_cdk import core, aws_ec2 as ec2

app = core.App()
vpc_id = "vpc-08c949884e65e3b6f"
#env = core.Environment(region="us-west-1",account="000")
env = {"region": "us-west-1", "account": "000"}


class addvpc(core.Stack):
    def __init__(
        self,
        scope: core.Construct,
        id: str,
        env: dict,
        **kwargs,
    ) -> None:
        super().__init__(scope, id, **kwargs)
        self.vpc = ec2.Vpc(
            self,
            id=f"Just-a-test-vpc",
            max_azs=2,
            cidr="192.168.0.0/16",
        )


class VPCImport2(core.Stack):
    def __init__(
        self, scope: core.Construct, id: str, env: dict, vpc_id: str, **kwargs
    ) -> None:
        super().__init__(scope, id, **kwargs)
        print(env)
        vpc = ec2.Vpc.from_lookup(
            self,
            "vpc_imported",
            is_default=False,
            vpc_id=vpc_id,
        )


vpc_stack = addvpc(scope=app, id="Addfakevpc", env=env)
import_vpc_stack = VPCImport2(scope=app, id="vpc-import", env=env, vpc_id=vpc_id)

app.synth()
```

### What did you expect to happen?
The VPC that was just created would get imported,

### What actually happened?

`  Error: Cannot retrieve value from context provider vpc-provider since account/region are not specified at the stack level. Either configure "env" with explicit account and region when you define your stack, or use the environment variables "CDK_DEFAULT_ACCOUNT" and "CDK_DEFAULT_REGION" to inherit environment information from the CLI (not recommended for production stacks)`

```
cdk list -v
...
Setting "CDK_DEFAULT_REGION" environment variable to us-west-1
Resolving default credentials
Retrieved account ID 000 from disk cache
Setting "CDK_DEFAULT_ACCOUNT" environment variable to 000
context: {
  'availability-zones:account=000:region=us-west-1': [ 'us-west-1a', 'us-west-1c' ],
....
```

### Environment

  - **CDK CLI Version**  : 1.8.1
  - **Framework Version:**
  - **Node.js Version:** v14.15.1
  - **OS               :** Debian Buster
  - **Language (Version):**  Python (3.9.1)

### Other

I feel like the error that is being thrown is not the real error but is some other underlyaing problem as I cant believe that this would stop working just for one certain function.

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by rix0rrr**
How the `env` parameter is supposed to be used is still confusing to
a lot of users.

Add uncommentable lines to the init templates showing and describing
the 3 alternatives.

Fixes #12321.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by rix0rrr** [comment]
Not merging yet -- would like to get some agreement on verbiage before I apply the same to all templates.

File: packages/aws-cdk/lib/init-templates/v1/app/typescript/bin/%name%.template.ts [hunk 1]
```diff
@@ -4,4 +4,16 @@ import * as cdk from '@aws-cdk/core';
 import { %name.PascalCased%Stack } from '../lib/%name%-stack';
 
 const app = new cdk.App();
-new %name.PascalCased%Stack(app, '%name.PascalCased%Stack');
+new %name.PascalCased%Stack(app, '%name.PascalCased%Stack', {
+  /* If you don't specify 'env', this stack will be environment-agnostic.
```
Comment (nija-at) [inline packages/aws-cdk/lib/init-templates/v1/app/typescript/bin/%name%.template.ts:8]: Other languages? v2 templates?

**Review by nija-at — CHANGES_REQUESTED** [review]

File: packages/aws-cdk/lib/init-templates/v1/app/typescript/bin/%name%.template.ts (hunk 1 above)
Comment (rix0rrr) [reply packages/aws-cdk/lib/init-templates/v1/app/typescript/bin/%name%.template.ts:8]: > Not merging yet -- would like to get some agreement on verbiage before I apply the same to all templates.

File: packages/aws-cdk/lib/init-templates/v1/app/typescript/bin/%name%.template.ts [hunk 2]
```diff
@@ -4,4 +4,16 @@ import * as cdk from '@aws-cdk/core';
...
+   * Account/Region-dependent features and context lookups will not work,
+   * but a single synthesized template can be deployed anywhere. */
```
Comment (nija-at) [inline packages/aws-cdk/lib/init-templates/v1/app/typescript/bin/%name%.template.ts:10]:
```suggestion
   * but a single synthesized template can be deployed anywhere.
   * See https://docs.aws.amazon.com/cdk/latest/guide/stacks.html to
   * learn more about environment agnostic stacks.
   * See https://docs.aws.amazon.com/cdk/latest/guide/context.html to
   * learn more about runtime contexts. */
```

**Review by nija-at — COMMENTED** [review]
I'm ok using this and the verbiage.

Just one comment.

**Review by nija-at — APPROVED** [review]

**Pull request merged**